### PR TITLE
Use warnings in checkflag on common interface

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
Instead of throwing a Julia error, the common interface will now warn. This makes Sundials compatible with the addon functionality like parameter estimation which can choose parameters that diverge. Before, this would make Sundials throw an error, which would error the higher level routine. Now it will return the current solution at the time of the error, which the parameter estimation techniques know how to handle.

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
Example that used to not work:

```julia
using DiffEqParamEstim, OrdinaryDiffEq, ParameterizedFunctions,
      DiffEqBase, RecursiveArrayTools
using Base.Test

# Here's the problem to solve

f = @ode_def_nohes LotkaVolterraTest begin
  dx = a*x - b*x*y
  dy = -c*y + d*x*y
end a=>1.5 b=1.0 c=3.0 d=1.0

u0 = [1.0;1.0]
tspan = (0.0,10.0)
prob = ODEProblem(f,u0,tspan)
sol = solve(prob,Tsit5())

# Generate random data based off of the known solution

t = collect(linspace(0,10,200))
randomized = [(sol(t[i]) + .01randn(2)) for i in 1:length(t)]
data = vecvec_to_mat(randomized)

fit = lm_fit(prob,t,vec(data),[1.0],CVODE_BDF(),show_trace=true,lambda=10000.0)

import Optim
result = Optim.optimize(obj, [1.0], Optim.BFGS())
```

Optim's BFGS goes into a parameter regime where the Lotka-Volterra equations diverge. This makes it so that way Sundials fails. The previous failure would cause the whole parameter estimation routine to exit. However, now it just calculates a high cost in the cost function, so Optim continues.

Down the line, the warning mechanism will be replaced with a verbosity command to turn on/off warnings and retcodes. This will be the standard DiffEq setup, but the retcodes are not implemented yet so we'll stick with this for now.

**Comment by ChrisRackauckas** [comment]
This PR is built off of the branch from #113.

File: src/common.jl [hunk 1]
```diff
@@ -106,6 +106,12 @@ function solve{uType,tType,isinplace,F,Method,LinearSolver}(
                                 save_ts[k], utmp, tout, CV_ONE_STEP)
                 push!(ures,copy(utmp))
                 push!(ts, tout...)
+                if flag != 0
+                  break
```
Comment (tkelman) [inline src/common.jl]: should use the same indentation style as the rest of the code

Comment (ChrisRackauckas) [inline src/common.jl]: My bad! I still need to find out how to make that automatic in Atom/Juno.

Comment (tkelman) [inline src/common.jl]: Juno should really be setting 4 spaces as default for Julia code.

File: src/common.jl [hunk 2]
```diff
@@ -118,13 +124,19 @@ function solve{uType,tType,isinplace,F,Method,LinearSolver}(
                 push!(ures,copy(utmp))
```
Comment (miguelraz) [inline src/common.jl]: push(ures, copy(utmp))

**Comment by ChrisRackauckas** [comment]
Moved this to #116 so that way it's not on a branch dependent on #113

**Pull request closed without merging**